### PR TITLE
Sustainability section

### PR DESCRIPTION
**Pull request by matthewturk**
This PR represents the first draft of the sustainability section.  I've tried to focus on things that are yt-specific, and only lightly discuss the broader topics related to sustainability of software packages.

### CODE REVIEW

**Comment by matthewturk** [comment]
I need to add a discussion of bug fixes.

File: content/70.sustainability.md [hunk 1]
```diff
@@ -1 +1,85 @@
...
+For the purposes of describing yt's approach to sustainability, we will use these as path markers but not restrictions.
+
+yt is supported through both peer-production and grant-funded development.
+At times, this can pose challenges; the requirements imposed by grant-funded development naturally concentrates decision making, but within the yt community we have (so far) navigated this through deep engagement in community processes and interaction.
+However, at the risk of belaboring a point that has been well-explored elsewhere, grant-funded development has traditionally focused on "new" or "innovative" development, rather than maintenance of a software project.
+
+A tension exists, however, between support of an existing project and the support of new projects in an ecosystem.
+By supporting an existing project, resources can tend to become concentrated; conversely, if a project supports a broader research agenda, that resource concentration can result in greater effort-multipliers for individuals who utilize the project.
+We're aware of this tension in yt; in fact, while yt has been grant-supported, most of the grant development has gone to a very few number of groups.
```
Comment (neutrinoceros) [inline content/70.sustainability.md:13]:
```suggestion
We're aware of this tension in yt; in fact, while yt has been grant-supported, most of the grant development has gone to a very small number of groups.
```

File: content/70.sustainability.md [hunk 2]
```diff
@@ -1 +1,85 @@
...
+This grant funding has been provided through the National Science Foundation, the Gordon and Betty Moore Foundation, the Department of Energy, the Chan Zuckerberg Initiative and other sources. [@doi:10.6084/m9.figshare.2061465.v1, [@doi:10.6084/m9.figshare.909413.v1], [@doi:10.5281/zenodo.4158589].
+Grants have supported the development of new features, including specific functionality for analysis routines and support for non-astronomical domains.
+
+Into each of these grants has been explicit support for community building, constituted by the development of documentation, videos, and tutorials, as well as mentoring of new contributors and shepherding the growth of the project through code review and issue management.
+While this does provide support for individuals who can provide dedicated, thoughtful attention to code review and bug reports, it is, quite frankly, insufficient without the engagement of non-funded community members who contribute their time and energy.
+
```
Comment (neutrinoceros) [inline content/70.sustainability.md:19]: ❤️

File: content/70.sustainability.md [hunk 3]
```diff
@@ -1 +1,85 @@
...
+It is tough to estimate the cost of bug fixes, in terms of labor or in terms of community member attrition, but @doi:10.1007/s11219-021-09564-z suggests that large, established projects on github may spend up to 40% of their time on bugfixes rather than new features.
+While we have not been able to verify this, examining the cadence of bug fix releases versus new feature releases suggests that this is indeed the case for yt as well.
+
+Maintenance and bug fixes are critical to developing yt as a community-focused code.
+The damage that can be done by ignoring a bug report, dismissing a suggestion, or outright rejecting proposed fixes is real, and can have measurable impact on community cohesion and growth.
+As such, in the yt development process we strive to be as accommodating and welcoming as possible, while still balancing that against the very real costs that responding to bug fixes, feature requests, and providing helpful insight with issues that may be covered by the documentation or that are known bugs.
```
Comment (neutrinoceros) [inline content/70.sustainability.md:52]: I *think* this sentence isn't finished ? I can't seem to get the hang of it

File: content/70.sustainability.md [hunk 4]
```diff
@@ -1 +1,85 @@
...
+This is a challenge, and one that benefits from having dedicated maintainers, or at least dedicated maintenance time from the developer community.
+Furthermore, we have found that regular, collaborative co-working sessions can ease this burden, although they occasionally veer into too light of code review or constant reiteration of "standing" issues.
+
+### Ecosystem Maintenance
+
+yt exists inside a larger ecosystem of packages and infrastructure, composed of the Python language, the Scientific Python community, as well as the computational science communities (predominantly astrophysics).
+Each of these communities, much like yt itself, has its own set of priorities, needs, and development schedules.
+Changes in systems that yt depends on may require changes in yt; changes in what other systems expect from yt may likewise require changes in yt.
+
+One of the more notable shifts in recent years has been in how python libraries and applications are distributed.
```
Comment (neutrinoceros) [inline content/70.sustainability.md:62]: A general check for case consistency is probably warranted
```suggestion
One of the more notable shifts in recent years has been in how Python libraries and applications are distributed.
```

File: content/70.sustainability.md [hunk 5]
```diff
@@ -1 +1,85 @@
...
+At the time of the first yt method paper (which, granted, is more than a decade prior to this one) yt was distributed typically in source form with its entire dependency stack to provide a uniform experience on supercomputer installations.
```
Comment (neutrinoceros) [inline content/70.sustainability.md:63]: Why you hitting yourself ?
```suggestion
At the time of the first yt method paper, yt was distributed typically in source form with its entire dependency stack to provide a uniform experience on supercomputer installations.
```

File: content/70.sustainability.md [hunk 6]
```diff
@@ -1 +1,85 @@
...
+This was, at the time, "necessary" because Python was not as widely used in the computational science and astrophysics communities as it is now, and the versions of various libraries were often far out of date.
+In the intervening time, however, a number of shifts have occurred, including the advent of conda (and conda-forge), wheels, wide-scale adoption of python in the scientific community, and considerable standardization and improvement of package metadata and installation methods.
```
Comment (neutrinoceros) [inline content/70.sustainability.md:65]:
```suggestion
In the intervening time, however, a number of shifts have occurred, including the advent of conda (and conda-forge), wheels, wide-scale adoption of Python in the scientific community, and considerable standardization and improvement of package metadata and installation methods.
```

File: content/70.sustainability.md [hunk 7]
```diff
@@ -1 +1,85 @@
...
+The balance with these changes can be difficult to strike, to minimize disruption to existing developers and users while still expanding the accessibility of the library to others.
+
+### Enjoyment Maximization
+
+Another aspect of sustainability is that for many people, developing software can be *fun*.
+This is certainly true for many members of the yt community -- sometimes in ways that align with the goals and requirements of academic research, and sometimes in ways that do not directly advance those goals.
+For instance, the initial development of the software volume renderer was not aligned with immediate research goals, but it has gone on to support synthetic observations and to foster much broader research and education opportunities.
+
+While development for the joy of developing can be an exciting and helpful process, it is also something that can require a delicate balance of attention and engagement from the rest of the community.
+Accepting changes into the yt codebase requires careful review, and dedication of resources (real and intellectual) that may already be allocated elsewhere.
```
Comment (neutrinoceros) [inline content/70.sustainability.md:81]: opposing "real" to "intellectual" feels slightly offensive, maybe ?
```suggestion
Accepting changes into the yt codebase requires careful review, and dedication of resources (physical and intellectual) that may already be allocated elsewhere.
```

**Review by neutrinoceros — COMMENTED** [review]
a couple minor comments, but otherwise I fully agree with everything you're adding here, thank you !

File: content/70.sustainability.md (hunk 3 above)
Comment (matthewturk) [reply content/70.sustainability.md]: Can you take another look?

**Pull request merged**